### ISSUE TITLE
Sugestion - Add methods to check if file is Gzipped

### ISSUE THREAD

**Issue by Zasuk** [mattt/Godzippa#16]
Adding some methods in the categories to check if the file is gzipped or not is really useful.
For the file manager i think it should load only the first bytes to avoid loading the full file to memory.

**How to check if is gzip:**
https://github.com/nicklockwood/GZIP/blob/7f40cad1de67b75e372704035ed2cdca44a4dc52/GZIP/NSData%2BGZIP.m#L130
or
https://filesignatures.net/index.php?search=GZ&mode=EXT
or
https://stackoverflow.com/a/19127748

**Comment by mattt** [comment]
I agree! Thanks for the great suggestion, @Zasuk. This was added by 9106fc332358da997aa89b937ad4f978c72eeae5 and is now available in 2.1.0.

**Comment by Zasuk** [comment]
Thanks.

for the FileManager extension i'm using some basic methods in my app...
The readFileData needs some extra error checking but from my tests it works.


```
- (NSData *)readFileData:(NSString *)sourcePath numberOfBytes:(NSUInteger)bytesToRead {
    
    NSInputStream *stream = [[NSInputStream alloc] initWithFileAtPath:sourcePath];
    NSMutableData *data = [NSMutableData data];
    [stream open];
    uint8_t buffer[bytesToRead];
    
    NSInteger result = [stream read:buffer maxLength:bytesToRead];
    [data appendBytes:buffer length:result];
    
    [stream close];
    
    if (data.length >= bytesToRead) {
        return data;
    }
    
    return nil;
}

// https://www.garykessler.net/library/file_sigs.html
- (BOOL)isGzip:(NSString *)sourcePath {
    
    NSData *data = [self readFileData:sourcePath numberOfBytes:3];
    if (data) {
        const UInt8 *bytes = (const UInt8 *)data.bytes;
        if(bytes[0] == 0x1f && bytes[1] == 0x8b && bytes[2] == 0x08) {
            return YES;
        }
    }
    
    return NO;
}
```
